### ISSUE TITLE
Can not make widget (checkbox) to read-only by adding flags PDF_FIELD_IS_READ_ONLY

### ISSUE THREAD

**Issue by remove158** [pymupdf/PyMuPDF#2350]
_**Please provide all mandatory information!**_

## Describe the bug (mandatory)
Can not make widget in types of (PDF_WIDGET_TYPE_CHECKBOX, PDF_WIDGET_TYPE_RADIOBUTTON) to read-only by adding flags `widget.field_flags |= fitz.PDF_FIELD_IS_READ_ONLY`  

Summary
Read-only is not work for checkboxes (bug).Moreover The value of the checkbox is missing after mark as read-only. Anyone has work-around solution ?

same with https://github.com/pymupdf/PyMuPDF/issues/2278 that mark as closed (but still didn't work yet)
## To Reproduce (mandatory)
```py
doc = fitz.open(name)
for page in doc:
    widgets = page.widgets()
    for widget in widgets :
        widget.field_flags |= fitz.PDF_FIELD_IS_READ_ONLY
        widget.update()
 ```

## Expected behavior (optional)
I want to make any widgets read-only by adding flags `fitz.PDF_FIELD_IS_READ_ONLY`

## Screenshots (optional)
![image](https://user-images.githubusercontent.com/23172564/223202070-1bd265f6-b58f-47e8-946f-db01f38cd38e.png)

## Your configuration (mandatory)
 - Operating system, potentially version, and bitness window 11 education 22H2
 - Python version, bitness 3.11.2
 - PyMuPDF version, installation method ^1.22.0

**Comment by remove158** [comment]
I tryna use this to work-around and it didn't work

```python
doc = fitz.open(name)
for page in doc:
    widgets = page.widgets()
    for widget in widgets :
        widget.field_flags |= fitz.PDF_FIELD_IS_READ_ONLY
        doc.xref_set_key(widget.xref, "Ff", str(widget.field_flags))```

**Comment by julian-smith-artifex-com** [comment]
Fixed in PyMuPDF-1.22.1.
